internal/context/context.go

```diff
@@ -20,6 +21,26 @@ func (k *contextKey) String() string {
 	return k.Name
 }
 
+type RPCContextData struct {
+	Method string
+	URI    string
+}
+
+func (rpcc RPCContextData) Copy() RPCContextData {
+	return RPCContextData{
+		Method: rpcc.Method,
+		URI:    rpcc.URI,
+	}
+}
+
+func (rpcc RPCContextData) IsSingleFileChange() (string, bool) {
+	if rpcc.Method == "textDocument/didChange" {
+		return path.Base(rpcc.URI), true
+	}
+
+	return "", false
+}
```

Comment:
I was originally under the impression that we'd want to check for multiple RPC methods but if we're only basically checking for `textDocument/didChange`, would it make more sense to just move this logic into the job and get rid of it here? It feels like unnecessary abstraction layer now.